Fix for War missing deaths by TNT + minor menu & message updates

Iniquit opened the pull request.

Iniquit committed:
Update message.properties

Added more pvp.kill.verbs; minor punctuation & phrasing edits. Added message notifying player they can sneak in spawn to change classes to the "inventory in storage" message upon joining warzones.

Iniquit committed:
Add check for Primed TNT to onEntityDamage handler

Exempt PrimedTNT entities from the onEntityDamage handler's "caused by entity" exclusion clause and allow them to be handled here since they are not handled by the onEntityDamageByEntity "PVP" handler

Iniquit committed:
Minor menu improvements

Hide default item attribute lore (like on weapons and armor) on items in menus. 
"Light up" ender pearls into ender eyes for warzone icons in the JoinZoneUI when they're occupied by 1 or more players.

james9909 commented on war/src/main/java/com/tommytony/war/listeners/WarEntityListener.java:
```diff
@@ -280,6 +278,65 @@ public void onEntityDamageByEntity(EntityDamageByEntityEvent event) {
+        if (event instanceof EntityDamageByEntityEvent) {
+
+            //and it's primed TNT, let it through
+            if (((EntityDamageByEntityEvent) event).getDamager() instanceof TNTPrimed){}
```
I think we could make this a bit cleaner by not having this explicit whitelisting behavior and instead add a branch within `onEntityDamageByEntity` and check if `attacker instanceof TNTPrimed`.

Looking into this a little further, it seems like we can get the entity who set off the TNT, using [getSource](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/TNTPrimed.html#getSource--), allowing us to credit attackers who use TNT to kill an enemy.

Iniquit replied:
Sounds good! I've made a new commit that moves the check under `onEntityDamageByEntity` like you described and attempts to get credit the player who lit the TNT.

It looks like the`EntityDamageByEntityEvent`  filter in the `onEntityDamage` handler 
```
   if (event instanceof EntityDamageByEntityEvent) {
            return;
        }
```
will still prevent War from catching deaths to non-living entities not originating with a player (like summoned Primed TNT, or arrows shot from a dispenser), since `onEntityDamage` looks only for damage from non-entities and `onEntityDamageByEntityEvent` looks only for 'PVP' damage, where the attacker is a player, or an entity that can be attributed to a player.

Death handling seems to work as expected without it. Is this okay to remove?

james9909 replied:
Sure, we can remove that check if it doesn't do anything. I didn't add that code, but it seems like it was protecting against certain death duplications since `EntityDamageByEntityEvent` is a subclass of `EntityDamageEvent`, and I'm not too sure what Spigots behavior with that is.

If you've tested all the scenarios for death and nothing seems out of the ordinary, then we should remove that check.

Iniquit replied:
It seemed fine for the scenarios I tested, but I definitely didn't hit them all-- I'll go through more methodically and let you know if I catch anything.

Do you think it would make sense to eventually unify these two damage handlers to prevent any kind of duplication? `onEntityDamage` and then have it split using `if (event instanceof EntityDamageByEntityEvent)`, or something.

james9909 replied:
Yeah, that's a great idea! We can do that separately after we merge in this PR.

Once you've tested everything out, make another comment and we can merge this in.

james9909 reviewed (changes requested):
Left a comment about where I feel like the `TNTPrimed` logic should be moved too, but the rest of it looks good!

Iniquit committed:
Move check for Primed TNT to onEntityDamageByEntity handler

Also remove "EntityDamageByEntity" filter from onEntityDamage handler because entities that were not shot or lit by a player (ex.: summoned primed TNT, arrows shot from a dispenser) could still kill without the death being handled by either handler
